examples/i18n-routing/components/locale-switcher.js

```diff
@@ -0,0 +1,26 @@
+import { useRouter } from 'next/router'
+import Link from 'next/link'
+
+export default function LocaleSwitcher() {
+  const router = useRouter()
+  const { locales, locale: activeLocale } = router
+  const otherLocales = locales.filter((locale) => locale !== activeLocale)
+
+  return (
+    <div>
+      <p>Locale switcher:</p>
+      <ul>
+        {otherLocales.map((locale) => {
+          const { pathname, query } = router
+          return (
+            <li key={locale}>
+              <Link href={{ pathname, query }} locale={locale}>
```

Comment:
```suggestion
              <Link href={{ pathname, query }} as={asPath} locale={locale}>
```

The `asPath` should also be passed if only the locale is meant to be changed

Reply:
Thanks for the review. I thought the `as` prop is not needed any more for dynamic routes since [Next.js 9.5.3](https://nextjs.org/docs/api-reference/next/link#if-the-route-has-dynamic-segments). I also saw a number of commits to the docs removing the `as` prop from some examples due to this as well.

Reply:
It's not needed when you are passing the full URL directly e.g. `/blog/post-1` but if you are using the `pathname` from the router like this and you only want to change the `locale` you need to make sure to pass all of the router information. 

If you are on `/blog/post-1` and you change the locale here it would try navigation to `/blog/[slug]` since the `asPath` isn't provided which would then be interpolated via the query but this would fail for cases where rewrites are used as well.

Reply:
Thanks for the info. Is this mentioned anywhere in the docs currently? if not can you point me to the relevant code & test cases and I will try adding information about this to the i18n docs page.

Reply:
I'm not sure changing the page like this is documented, it might be good to add this in the `i18n-routing` doc or `next-router` doc